src/dotnet/APIView/APIViewWeb/Repositories/PullRequestManager.cs

```diff
@@ -293,6 +293,14 @@ private ReviewModel CreateNewReview(PullRequestModel prModel)
                     var prevRevisionId = review.Revisions.Last().RevisionId;
                     review = await GetBaseLineReview(codeFile.Language, codeFile.PackageName, pullRequestModel, true);
                     review.ReviewId = pullRequestModel.ReviewId;
+                    //Remove previous revisions with revision ID.
+                    //Currently revision ID is getting duplicated when a PR api review is created for a brand new package.
+                    //In case of brand new package, we don't have any baseline from automatic review. So it uses previous PR api review as baseline and
+                    //below revision ID copy step makes duplicate revision IDs in such cases.
+                    //We should ensure that no revision exists in review with previous revision ID before we update new revision
+                    var revisionsToRetain = review.Revisions.Where(r => r.RevisionId != prevRevisionId);

```

Comment:
```suggestion
                    review.Revisions.RemoveAll(r => r.RevisionId == prevRevisionId);

```

Reply:
List is a custom List implementation and didn't have RemoveAll. I have added this method now and calling RemoveAll.

Reply:
Ah, I forgot about that.